packages/story-editor/src/components/floatingMenu/context/provider.js

```diff
@@ -25,8 +26,11 @@ import PropTypes from 'prop-types';
 import Context from './context';
 
 function FloatingMenuProvider({ handleDismiss, children }) {
+  const designMenuRef = useRef(null);
```

Comment:
💭 there's a ref to the design pane in [the inspector provider](https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/story-editor/src/components/inspector/inspectorProvider.js#L76). It's a different provider but since it's a reference to the design pane you could always grab it from there instead of needing to set up another one.

---

💭 But rather than checking where it's being rendered, we could follow the pattern of the [`Smart Popover` component](https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/design-system/src/components/contextMenu/styled.js#L55-L84) that checks its dimensions and renders itself in the correct place

Reply:
the `designMenu` is referring to the `floating menu` not the design pane. And really, if we just make `dock` `null`, it'll render where it is supposed to. since the anchor is in the floating menu. So I am not sure we even need a ref to the floating menu at all.

Reply:
i ended up removing this